core/src/main/scala/kafka/server/metadata/BrokerMetadataListener.scala

```diff
@@ -119,16 +119,24 @@ class BrokerMetadataListener(
       }
 
       _bytesSinceLastSnapshot = _bytesSinceLastSnapshot + results.numBytes
-      if (shouldSnapshot()) {
-        maybeStartSnapshot()
+      
+      val shouldTakeSnapshot: Option[String] = shouldSnapshot()
+      if (shouldTakeSnapshot.isDefined) {
+        maybeStartSnapshot(shouldTakeSnapshot.get)
       }
 
       _publisher.foreach(publish)
     }
   }
 
-  private def shouldSnapshot(): Boolean = {
-    (_bytesSinceLastSnapshot >= maxBytesBetweenSnapshots) || metadataVersionChanged()
+  private def shouldSnapshot(): Option[String] = {
+    if (_bytesSinceLastSnapshot >= maxBytesBetweenSnapshots) {
```

Comment:
I think it's possible that both these 2 conditions are satisfied at the same time, right? so maybe we should have another condition: "max bytes exceeded and metadata version changed", WDYT?

Reply:
Makes sense, I wasn't sure whether this scenario was possible or not. Haven't dug deep yet. Should've clarified this before, my bad.

Have made the changes to accommodate this.